mlflow/pyfunc/scoring_server/__init__.py

```diff
@@ -132,6 +137,74 @@ def infer_and_parse_json_input(json_input, schema: Schema = None):
         )
 
 
+def _decode_json_input(json_input):
+    """
+    :param json_input: A JSON-formatted string representation of TF serving input or a Pandas
+                       DataFrame, or a stream containing such a string representation.
+    :return: A dictionary representation of the JSON input.
+    """
+    if isinstance(json_input, dict):
+        decoded_input = json_input
+    else:
+        try:
+            decoded_input = json.loads(json_input)
+        except json.decoder.JSONDecodeError as ex:
+            raise MlflowException(
+                message=(
+                    "Failed to parse input from JSON. Ensure that input is a valid JSON"
+                    f" formatted string. Error: '{ex}'. Input: \n{json_input}\n"
+                ),
+                error_code=BAD_REQUEST,
+            )
+
+    if isinstance(decoded_input, dict):
+        return decoded_input
```

Comment:
we can remove this if we do the initial return
```suggestion
```

Reply:
It could come from 'json.loads(json_input)'

Reply:
it certainly can :) disregard!